### PR TITLE
[FEATURE] Explain locking API

### PR DESCRIPTION
**Pull request by sypets**
Related: #85

### CODE REVIEW

File: Documentation/ApiOverview/LockingApi/Index.rst [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+* SemaphoreLockStrategy: 25
+
+Examples
+--------
+
+Acquire and use an exclusive, blocking lock::
+
+   $lockFactory = GeneralUtility::makeInstance(LockFactory::class);
+
+   // createLocker will return an instantiation of class which implements
```
Comment (liayn) [inline Documentation/ApiOverview/LockingApi/Index.rst]: return an instance of

File: Documentation/ApiOverview/LockingApi/Index.rst [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+   // Here, we are asking for an exclusive, blocking lock. This is the default,
+   // so the second parameter could be omitted.
+   $locker = $lockFactory->createLocker('someId', LockingStrategyInterface::LOCK_CAPABILITY_EXCLUSIVE);
+
+   // now use the locker to lock something exclusively, this may block (wait) until lock is free, if it
+   // has been used already
+   $locker->acquire(LockingStrategyInterface::LOCK_CAPABILITY_EXCLUSIVE);
+
+   // do some work that required exclusive locking here ...
+   if ($locker) {
```
Comment (liayn) [inline Documentation/ApiOverview/LockingApi/Index.rst]: This is not correct. one has to the check the result from $locker->acquire

File: Documentation/ApiOverview/LockingApi/Index.rst [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+       // after you did your stuff, you should release
```
Comment (liayn) [inline Documentation/ApiOverview/LockingApi/Index.rst]: you must release

File: Documentation/ApiOverview/LockingApi/Index.rst [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+.. _locking-api-caveats:
+
+Caveats
+=======
+
+FileLockStrategy & NFS
+----------------------
+
+There was a problem with PHP `flock()
```
Comment (liayn) [inline Documentation/ApiOverview/LockingApi/Index.rst]: is a problem

**Comment by liayn** [comment]
excellent work

**Comment by sypets** [comment]
@liayn Thank you so much for your feedback. I'll push another patch for the "There was a problem" and do final proofreading.

**Comment by liayn** [comment]
IMO this can be merged.

**Comment by DanielSiepmann** [comment]
Thanks @sypets for providing the docs.
Thanks @liayn for providing the feedback.

**Comment by sypets** [comment]
@DanielSiepmann Thanks for merging!

**Pull request merged**